---
title: "01_Human_PANDA_TFmotif"
output: html_document
date: '2022-11-15'
---

# Generating own CIS-BP Tf-Motif input:

* Grabbing TF-Motif Input for Homo Sapiens (Human):
  * Accessed 221115 from Kimberly Glass Resources "PANDA-ready Motif Mappings" (https://sites.google.com/a/channing.harvard.edu/kimberlyglass/tools/resources)
  * The resource contains a Human Motif Scan (file name: Homo_sapiens_motifinfo.txt) for hg38 here that contains the TF and its MOTIF ID: https://drive.google.com/file/d/1MlPiAmOxu4mTd4nW9zMygzK1U9F8XB0M/view
  * Additionally a Prior Network (file name: hg38_reseq_-750_250_1e-4.txt) is provided that contains MOTIF IDs and the gene that they bind to here: https://drive.google.com/file/d/1Wmw64hZOIoIrdeE2XjUulmbFp2RArcFa/view
  * How motif mapping was generated by Glass: The mappings were produced by using FIMO to scan a genome for sets of  CIS-BP Single Species DNA-motifs curated in the MEME suite. Hits that met a significance less than 1e-4 were recorded and mapped to genes based on annotations downloaded from UCSC. Significant motif hits that fell within [-750,+250] of a gene's transcriptional start site were used to construct a prior regulatory network. 


Note: The TF Motif input is needed in order to construct string PPI inputs for the network analysis. This script precedes the 02_Human_TF_motif_enrichment.Rmd and 03_Human_PANDA_ppi.Rmd. The output generated here is needed as an input for that analysis. 

## Code: 

Load in libraries needed:
```{r}
library(stringr)
library(dplyr)
library(here)
library(styler)
library(lintr)
set.seed(2178)
```


Programatically read the .txt files into R (see notes and script in accompanying dry lab notebook referenced above). Need to map the MOTIF IDs in the prior network and their corresponding genes to the MOTIF IDs in the Human Motif Scan to create a final input dataset that contains the following columns: "gene", "TF", "1/0" where the 1 indicates the relationship being present (final input dataset should be all 1's)

```{r}
# load in the data from Glass Resource:
human_motif <- read.table(here("data/Homo_sapiens_motifinfo.txt"), header = TRUE) # has column headers
prior_network <- read.table(here("data/hg38_refseq_-750_250_1e-4.txt")) # does not have column headers
```

### Clean up human_motif input data:
```{r}
# remove the version numbers on Motif IDs from MotifID column:
MotifIDs <- str_split(human_motif$MotifID, "\\_", simplify = TRUE)[, 1] # grab list of IDs without version
human_motif$MotifID <- MotifIDs # set list of cleaned up IDs and MotifID column values

# remove parentheses and other characters from TF info so that TF only contains name of TF:
TFs <- str_split(human_motif$TF_Info, "\\_", simplify = TRUE)[, 1] # splits every column value out by separator and only grabs the gene names
TFs <- gsub("[()]", "", TFs) # removes genes from ()
human_motif$TF_Info <- TFs
dim(human_motif) # 733 total TFs, 2 columns 1 for TF_Info and 1 for MotifID. NOTE: there are three unknown genes that do not have a gene symbol (ENSG00000250542, ENSG00000235187, ENSG00000229544), may need to remove later if cause issues in network construction
```

### Add column names to prior network for mapping:
```{r}
colnames(prior_network) <- c("MotifID", "Motif", "interaction")
head(prior_network)
```


### Map motifs to TF using Motif IDs using innerjoin:
```{r}
hg38_TFmotifs <- inner_join(human_motif, prior_network, by = "MotifID") # generate final input data for PANDA

hg38_TFmotifs <- subset(hg38_TFmotifs, select = -c(MotifID)) # drop interaction information and Motif IDs, because all interactions are present (represented as a 1) and this information is not needed for PANDA input
write.table(hg38_TFmotifs, file = here("data/hg38_TFmotifs.txt"), sep = "\t") # save final input data to Cheaha for use in network construction
```

```{r}
# run style
style_file(here("src/panda_input_construction/01_Human_PANDA_TFmotif.Rmd"))
# lintr was run as well
```

```{r}
sessionInfo()
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lintr_3.0.2   styler_1.9.0  here_1.0.1    dplyr_1.1.0   stringr_1.5.0

loaded via a namespace (and not attached):
 [1] pillar_1.8.1      compiler_4.1.3    R.methodsS3_1.8.2 R.utils_2.12.2    remotes_2.4.2     tools_4.1.3       digest_0.6.31     lifecycle_1.0.3  
 [9] tibble_3.1.8      R.cache_0.16.0    pkgconfig_2.0.3   rlang_1.0.6       rex_1.2.1         cli_3.6.0         rstudioapi_0.14   xfun_0.37        
[17] cyclocomp_1.1.0   xml2_1.3.3        withr_2.5.0       knitr_1.42        generics_0.1.3    desc_1.4.2        vctrs_0.5.2       rprojroot_2.0.3  
[25] tidyselect_1.2.0  glue_1.6.2        R6_2.5.1          processx_3.8.0    fansi_1.0.4       purrr_1.0.1       callr_3.7.3       magrittr_2.0.3   
[33] ps_1.7.2          utf8_1.2.3        stringi_1.7.12    lazyeval_0.2.2    crayon_1.5.2      R.oo_1.25.0 

```{r}
proc.time
```
   user  system elapsed 
 22.517   2.664  29.073 

