---
title: "02_Human_TF_motif_enrichment"
output: html_document
date: '2023-01-23'
output:     
  html_document:
      toc: true
      toc_depth: 2
      toc_float: true
---

Goal: Reconstruct TF-motif input to include publicly available ChIP-seq data from MsigDB, interactions from TRRUST, and SIGNOR 
* Step 1: Load previously generated TF-motif input data for human and Setbp1 ChIP-seq data from GTRD
* Step 2: Filter SIGNOR data for higher confidence in interactions 
* Step 3: Enrich TF-motif with Setbp1 ChIP-seq data and save

TRRUST v.2.0 (2017) accessed January 2023
* Human: https://www.grnpedia.org/trrust/data/trrust_rawdata.human.tsv
TRRUST is a manually curated reference database of human and mouse TF-target interactions. 

SIGNOR 3.0 (2023) accessed January 2023
* Human and Mouse: https://signor.uniroma2.it/downloads.php 
SIGNOR is a database of manually annotated relationships between human proteins and known chemicals and their biological relevance. Interactions included were limited to only gene-protein interactions that are classified as "transcriptional regulation" in order to construct the TF-motif input data. SIGNOR scores range from 0.1 - 1.0 and were filtered to only include interactions with a score of at least 0.4. This threshold was based on previously published SIGNOR 3.0 score update, showing the proportion of components and scores, and whether their interaction is annotated to a pathway (https://www.syngoportal.org/convert) After filtration, all scores were converted to 1. This is because all other TF-motif inputs have a binary 1/0 scoring system where 1 indicates a reaction that is present and 0 an interaction that is not. This is done in an effort to avoid bias in network construction, weighting some interactions as more important than others so that expression is the main driver or regulation in PANDA construction. 

MSIGDB ChIP-seq data
* GTRD Setbp1 Target Genes: https://www.gsea-msigdb.org/gsea/msigdb/cards/SETBP1_TARGET_GENES.html
There are 188 interactions for human with Sebtp1 found in the GTRD ChIP-seq experiments for SETBP1.

# Setup
```{r loading-libraries, results='hide'}
library(readr)
library(dplyr)
library(here)
library(styler)
library(lintr)
set.seed(2178)
```

# Load input files

## Human
Note: The previous TF_motif input was generated in the 'Human_PANDA_TFmotif_02.Rmd'
```{r load-previous-TF-motif}
hu_motif <- read.table(file = here("data/hg38_TFmotifs.txt"), sep = "\t") # load in motif data
```

```{r load-human-SIGNOR}
signor_hu <- read_tsv(file = here("data/SIGNOR_all_human_data_23_01_23.tsv"), col_names = TRUE) # load in motif data
```

```{r load-human-TRRUST}
trrust_hu <- read_tsv(file = here("data/trrust_rawdata.human.tsv"), col_names = "X") # load in motif data
```

## Load in Setbp1 data from MSIGDB:
```{r}
SETBP1geneset_syngo <- read_csv(here("data/SETBP1genset_syngo.csv"))
```


# Create Human (Homo Sapiens) Input:
```{r filter-human-SIGNOR}
# filter for only "transcriptional regulation" and score >= 0.4, and proteins:
signor_hu <- signor_hu[signor_hu$MECHANISM == "transcriptional regulation", ] %>%
  filter(., rowSums(is.na(.)) != ncol(.)) %>%
  filter(., TYPEA == "protein") %>%
  filter(., TYPEB == "protein") %>%
  filter(., SCORE >= 0.4, )

# grab ENTITYA, ENTITYB, and SCORE columns (correspond to TF, Motif, and Score)
signor_hu <- signor_hu[, c(1, 5, 27)]
colnames(signor_hu) <- c("TF_Info", "Motif", "interaction")
signor_hu$interaction <- 1
```

```{r wrangle-human-TRRUST}
colnames(trrust_hu) <- c("TF_Info", "Motif", "mechanism", "PMID")
trrust_hu <- trrust_hu[, c(1:2)]
trrust_hu$interaction <- 1
```

```{r merge-all}
hu_motif_all <- unique(rbind(hu_motif, signor_hu) %>% rbind(., trrust_hu))
```

```{r manually-add-SETBP1-chipseq-human}
msig <- matrix(NA, nrow = 188, ncol = 3)
msig <- data.frame(msig)
colnames(msig) <- c("TF_Info", "Motif", "interaction")
msig$Motif <- SETBP1geneset_syngo$query # add in geneset data
msig$TF_Info <- "SETBP1"
msig$interaction <- 1

# merge
hu_motif_all <- unique(rbind(hu_motif_all, msig))
```

```{r save-human}
write.table(hu_motif_all, file = here("data/motif_inputs/hu_motif_all.txt"), sep = "\t")
```

# Style file and lint file
```{r}
# run style
style_file(here("src/panda_input_construction/02_Human_TF_motif_enrichment.Rmd"))
# lintr was run as well
```

# Processing time
```{r}
proc.time()
```
   user  system elapsed 
143.345  25.066 449.363

# Session information
```{r}
sessionInfo()
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] lintr_3.0.2  styler_1.9.0 here_1.0.1   dplyr_1.1.0  readr_2.1.4 

loaded via a namespace (and not attached):
 [1] pillar_1.8.1      compiler_4.1.3    R.methodsS3_1.8.2 R.utils_2.12.2    remotes_2.4.2     tools_4.1.3       bit_4.0.5         digest_0.6.31    
 [9] evaluate_0.20     lifecycle_1.0.3   tibble_3.1.8      R.cache_0.16.0    pkgconfig_2.0.3   rlang_1.0.6       rex_1.2.1         cli_3.6.0        
[17] rstudioapi_0.14   yaml_2.3.7        parallel_4.1.3    xfun_0.37         fastmap_1.1.1     cyclocomp_1.1.0   xml2_1.3.3        withr_2.5.0      
[25] knitr_1.42        generics_0.1.3    desc_1.4.2        vctrs_0.5.2       hms_1.1.2         bit64_4.0.5       rprojroot_2.0.3   tidyselect_1.2.0 
[33] glue_1.6.2        R6_2.5.1          processx_3.8.0    fansi_1.0.4       vroom_1.6.1       rmarkdown_2.20    tzdb_0.3.0        purrr_1.0.1      
[41] callr_3.7.3       magrittr_2.0.3    htmltools_0.5.4   ps_1.7.2          ellipsis_0.3.2    utf8_1.2.3        lazyeval_0.2.2    crayon_1.5.2     
[49] R.oo_1.25.0  

