---
title: "03_Human_PANDA_ppi"
output: html_document
date: '2022-11-15'
---
# Overview: 
* Generating PPI Input for Human:
  * STRING:
    * The NetZoo package has a nice wrapper function here: https://netzoo.github.io/netZooR/reference/source.PPI.html, however, one cannot change the filtering threshold using this package. 
    * The stringDB package is used to pull in the latest version of string (v11.5, accessed May 2023)
    * All that is needed is for the user to specify the input query TFs that an individual would like to search for in the string database. 
    * Then the species.index must be specified, here I use the human NCBI taxon-id (human - 9606; https://string-db.org/cgi/organisms).
  * Input TFs were obtained from the TF Motif input constructed for both species and enriched for additional databases to include CIS-BP, TRRUST, SIGNOR, and MSIGDB SETBP1 ChIP-seq data. The files are called 'hu_motif_all.txt' and 'mus_motif_all.txt'
  
* Input data structure:
  * Column 1: TF_Info (TFs)
  * Column 2: Motif (Genes TF bind to on the DNA)
  * Column 3: Interaction (score, in this case a 1 for present or 0 for not present)
  
* PPI function parameters:
  * species_code: obtained from string-db organism codes (example: 9606)
  * string_version: version of string database user wants to construct PPI from (example: 11.5)
  * score_cutoff: Threshold scores go from 0 - 1000 and represent the confidence in the interactions present based on literature and experimental data. The lower the threshold, the less stringent. For example a score of 500 indicates every second interaction might be a false positive. 
    * (https://string-db.org/cgi/info)
  * output: name of output object

# Code: 

## loading in libraries
```{r}
library(STRINGdb)
library(stringr)
library(styler)
library(here)
source(here("src/functions.R"))
set.seed(2178)
```

## loading in the TF-motif input data (needed as input to construct PPI for both species):
These inputs were generated in the 02_Human_TF_motif_enrichment.Rmd
```{r}
hu_motif_all <- read.table(here("data/hu_motif_all.txt"),
  sep = "\t")
```


# generate human PPI input 

## constructing unthresholded human ppi
```{r constructing-human-ppi}
hg38_ppi <- ppi(species_code = 9606, score_cutoff = 0, string_version = 11.5, data = hu_motif_all)
```

## saving the final human PPI input 
```{r}
write.table(hg38_ppi, file = here("data/hg38_ppi.txt"), sep = "\t") # save final input data to Cheaha for use in network construction
```


```{r}
sessionInfo()
```
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] here_1.0.1     styler_1.9.0   stringr_1.5.0  STRINGdb_2.6.5

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10        compiler_4.1.3     RColorBrewer_1.1-3 plyr_1.8.8         R.utils_2.12.2     R.methodsS3_1.8.2  bitops_1.0-7       tools_4.1.3       
 [9] digest_0.6.31      bit_4.0.5          R.cache_0.16.0     lifecycle_1.0.3    RSQLite_2.3.0      memoise_2.0.1      pkgconfig_2.0.3    png_0.1-8         
[17] rlang_1.0.6        igraph_1.4.1       DBI_1.1.3          cli_3.6.0          rstudioapi_0.14    xfun_0.37          proto_1.0.0        fastmap_1.1.1     
[25] knitr_1.42         vctrs_0.5.2        gtools_3.9.4       caTools_1.18.2     rprojroot_2.0.3    bit64_4.0.5        glue_1.6.2         sqldf_0.4-11      
[33] plotrix_3.8-2      hash_2.2.6.2       gsubfn_0.7         purrr_1.0.1        blob_1.2.3         magrittr_2.0.3     gplots_3.1.3       KernSmooth_2.23-20
[41] stringi_1.7.12     RCurl_1.98-1.10    cachem_1.0.7       chron_2.3-59       R.oo_1.25.0       

```{r}
proc.time()
```
   user  system elapsed 
 88.319   8.218 208.093
 
```{r}
# run style
style_file(here("src/network_scripts/input_construction/Human_PANDA_ppi.Rmd"))
# lintr was run as well
```

