---
title: "tissue_specificity_pkd_gex"
author: "Lizzy Wilk"
date: "2023-04-19"
output:
    html_document:
      toc: true
      toc_depth: 4
      toc_float: true

---

load in packages
```{r}
library(tidyverse)
```

##Purpose  
This R markdown is for exploration of tissue-specific gene expression of PKD-associated genes using pre-calculated gene specificity from (Sonawane et al.)[https://www.sciencedirect.com/science/article/pii/S2211124717314183?via%3Dihub]
   
##Data
read in PKD-associated gene csv
```{r}
pkd_genes <- read.csv("./data/pkd_associated_genes.csv")
#name alteration to find matches
pkd_genes[8,1] <- "HNF1B" #"HNF1-beta" to HNF1B
```

Contents of RData file used for this R markdown:  
__expTS:__ a 30,243 by 38 matrix including, for each gene and each tissue, information on whether the gene is expressed in a tissue-specific manner in that tissue (1) or not (0).  
__genes:__ a 30,243 by 4 data.frame that includes annotation information (Symbol) for Ensembl gene IDs (Name). This data.frame also includes information on whether genes are also transcription factors (AlsoTF), with options: no, yes/motif (TF with a known DNA-binding motif) yes/nomotif (TF without a known DNA-binding motif). In addition, the multiplicity of the gene (Multiplicity) is given. 
  
load in RData file from Sonawane et al.
```{r}
load("./data/GTEx_PANAD_Tissues.Rdata")
```

##Tissue Specific Expression of PKD-Associated Genes
get annotation info for PKD-associated genes
```{r}
pkd_gene_anno <- dplyr::inner_join(pkd_genes, genes, by = c("gene" = "Symbol"))
#pkd-associated genes w/o availability
setdiff(pkd_genes$gene, pkd_gene_anno$gene)
```
 
filter tissue-specific expression for pkd-associated genes
```{r}
pkd_expTS <- expTS %>% as.data.frame(.) %>% 
  rownames_to_column(., var = "Name") %>% 
  dplyr::inner_join(., pkd_gene_anno, by = "Name")
```

save full annotation/tissue specific expression value table
```{r}
summary(pkd_expTS)
write.csv(pkd_expTS, "./results/pkd_expTS_fullannotation.csv")

#ComplexHeatmap::Heatmap(pkd_expTS[,2:39], row_labels = pkd_expTS$gene, column_labels = colnames(pkd_expTS)[2:39])
```

#### Filtering
remove tissues w/o any specific expression for pkd-associated genes
```{r}
tis_var <- pkd_expTS %>% summarise_if(is.numeric, var) %>% select(., !(Multiplicity))
#tis_keep <- colnames(tis_var[tis_var > 0])
tis_keep <- tis_var[,tis_var!=0]

pkd_expTS_min <- pkd_expTS %>% column_to_rownames(., var = "gene") %>% select(., c(colnames(tis_keep)))
  #pkd_expTS[colnames(pkd_expTS) %in% colnames(tis_keep),]
```

save table of tissue specific pkd-associated gene expression, filtered for tissues w/ specificity for at least one pkd-associated gene
```{r}
write.csv(pkd_expTS_min, "./results/pkd_expTS_filtered.csv")
```

## Visualize
annotation
```{r}

# row_ha <- ComplexHeatmap::rowAnnotation(Disease = pkd_gene_anno$disease, 
#                                         #Multiplicity = pkd_gene_anno$Multiplicity,
#                                         #colors
#                                         col = list(
#                                           Disease = c(unique(pkd_gene_anno$disease)[1] = viridis::rocket(6)[1],
#                                                       unique(pkd_gene_anno$disease)[2] = viridis::rocket(6)[2],
#                                                       unique(pkd_gene_anno$disease)[3] = viridis::rocket(6)[3], 
#                                                       unique(pkd_gene_anno$disease)[4] = viridis::rocket(6)[4],
#                                                       unique(pkd_gene_anno$disease)[5] = viridis::rocket(6)[5],
#                                                       unique(pkd_gene_anno$disease)[6] = viridis::rocket(6)[6]
#                                                       ))) 
pkd_dis <- pkd_gene_anno$disease
names(pkd_dis) <- pkd_gene_anno$gene

pkd_mult <- pkd_gene_anno$Multiplicity
names(pkd_mult) <- pkd_gene_anno$gene

row_ha <- ComplexHeatmap::rowAnnotation(Disease = pkd_dis, 
                                        Multiplicity = pkd_mult,
                                        #colors
                                        col = list(
                                          Disease = c("ADPKD" = viridis::cividis(6)[1],
                                                      "ARPKD" = viridis::cividis(6)[2],
                                                      "TSC" = viridis::cividis(6)[3], 
                                                      "HNF1betaDiease" = viridis::cividis(6)[4],
                                                      "NPHP" = viridis::cividis(6)[5],
                                                      "VonHippel-Lindau" = viridis::cividis(6)[6]
                                                      ))) 
row_anno <- pkd_gene_anno %>% column_to_rownames(., var = "gene") %>% reorder(., desc(Multiplicity)) %>% select(., Disease = disease, TF = AlsoTF, Multiplicity = Multiplicity)
row_ha <- ComplexHeatmap::rowAnnotation(df = row_anno)
                                        #, Multiplicity = c()))
# auto_ha <- function(feature, featurename){
#   row_ha <- NULL
#   palsize <- length(unique(feature))
#   for(i in 1:length(unique(feature))){
#    temp <- ComplexHeatmap::rowAnnotation(featurename = feature, 
#                                         #Multiplicity = pkd_gene_anno$Multiplicity,
#                                         #colors
#                                         col = list(featurename = c(paste(unique(feature)[i]) = viridis::rocket(palsize)[i])))
#    row_ha <- cbind(row_ha, temp)
#   }
# }
```

####Heatmap
```{r}
png("./results/pkd_expTS_diseaseanno_heatmap.png", width = 25, height = 30, units = "cm", res = 300)
set.seed(1)
ComplexHeatmap::Heatmap(as.matrix(pkd_expTS_min), row_labels = rownames(pkd_expTS_min), column_labels = colnames(pkd_expTS_min), right_annotation = row_ha) ##annotation order is wrong!!

dev.off()
```


#### Versions  
```{r}
R.Version()
```


```{r}
installed.packages()[names(sessionInfo()$otherPkgs), "Version"]
```
  
